### PR TITLE
refactor: Move metrics to internal package

### PR DESCRIPTION
**Pull request by ruanxin**
refactoring metrics related code into internal package.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -51,8 +51,7 @@ import (
 	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
 	"github.com/kyma-project/lifecycle-manager/internal"
 	"github.com/kyma-project/lifecycle-manager/internal/controller"
-	kymametrics "github.com/kyma-project/lifecycle-manager/internal/controller/kyma/metrics"   //nolint:importas // ad-hoc import collision resolution
-	purgemetrics "github.com/kyma-project/lifecycle-manager/internal/controller/purge/metrics" //nolint:importas // ad-hoc import collision resolution
+	"github.com/kyma-project/lifecycle-manager/internal/pkg/metrics"
```
Comment (lindnerby) [inline cmd/main.go:54]: Please the remove the `pkg` parent,  it's either `pkg` or `internal`. `metrics` is a nice enough package name for this

**Review by lindnerby — CHANGES_REQUESTED** [review]

File: cmd/main.go (hunk 1 above)
Comment (ruanxin) [reply cmd/main.go:54]: what do you mean the `pkg` parent?

Comment (lindnerby) [reply cmd/main.go:54]: What's the intent of creating a `pkg` subpackage within `internal`?

Comment (lindnerby) [reply cmd/main.go:54]: As of clarification we can indicate shared internal package to put them in `internal/pkg` 👍

**Review by lindnerby — APPROVED** [review]

**Pull request merged**